### PR TITLE
Redefine the simdlen semantic

### PR DESCRIPTION
**Pull request by EikanWang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89274
* __->__ #89263

This PR is targeting to automatically enable vectorization optimization for TorchInductor. It refined the semantics of `config.cpp.simdlen`. 

Originally, `None` means to disable vectorization while a specific value means the number of elements to be vectorized once time. But it depends on the data. Regarding 256bit SVE/SIMD ISA for ARM and X86, the `simdlen` should be 16 for Float while 32 for BFloat. Hence, this PR defined the `simdlen` as the bit width. The detailed semantics are as follows.

- **_simdlen = None_**: Automatically determine the SIMD bit width. Detect HW information and pick the proper vectorization ISA. Specific for X86, the priority of AVX512 is higher than AVX2.
- **_simdlen <=1_**: Explicitly disable SIMD
- **_simdlen > 1_**: Explicitly specify the SIMD bit width. It equals the disabled semantic if the bit width does not match the ISA width.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: test/inductor/test_minifier.py [hunk 1]
```diff
@@ -60,8 +60,10 @@ def _gen_codegen_fn_patch_code(self, old_fn_name, new_fn_code, device):
             patch_code = f"""\
 import torch._inductor.codegen.{"cpp" if device == "cpu" else "triton"} as codegen
 overrides = codegen.{"CppOverrides" if device == "cpu" else "TritonOverrides"}
+vec_overrides = codegen.{"CppVecOverrides" if device == "cpu" else "TritonOverrides"}
 {new_fn_code}
 overrides.{old_fn_name} = staticmethod({new_fn_name})
+vec_overrides.{old_fn_name} = staticmethod({new_fn_name})
 """
         return f"""\
 {patch_code}
```
Comment (EikanWang) [inline test/inductor/test_minifier.py:69]: The original PR https://github.com/pytorch/pytorch/pull/88482 was reverted due to this new added test case does not override the vectorization Overrides to test accuracy. Compared with the original PR, the only change is the test case polishment.

**Comment by EikanWang** [comment]
Just reland the PR.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]
